samples/iolib/src/main/cpp/player/SampleBuffer.cpp

```diff
@@ -18,24 +18,22 @@
 
 #include "io/wav/WavStreamReader.h"
 
-using namespace parselib;
-
 namespace iolib {
 
-void SampleBuffer::loadSampleData(WavStreamReader* reader) {
+void SampleBuffer::loadSampleData(parselib::WavStreamReader* reader) {
     mAudioProperties.channelCount = reader->getNumChannels();
     mAudioProperties.sampleRate = reader->getSampleRate();
 
     reader->positionToAudio();
 
     mNumSampleFrames = reader->getNumSampleFrames() * reader->getNumChannels();
-    mSampleData = new float[mNumSampleFrames];
-    reader->getDataFloat(mSampleData, reader->getNumSampleFrames());
+    mSampleData = std::make_shared<float*>(new float[mNumSampleFrames]);
```

Comment:
Should this be make_shared<float[]>(mNumSamples) ?

Reply:
I don't think so. The std::shared_ptr<> wraps a float*, not a float[]. Perhaps Mikhail can chime in on this?

Reply:
Well, prior to C++17 dealing with shared_ptrs to arrays was tricky because there was no dedicated support for that (unlike what unique_ptr has), and the culprit is that if you allocate an array using 'new[]' operator, you **must** delete it with 'delete[]' operator (this is what unique_ptr does). So prior to C++17 you have to provide a custom deleter if you use shared_ptr with a C array.

Since C++17 this issue has been fixed in the language and now it's OK to use shared_ptr with C arrays, and even support for 'operator[]' to access array elements was added to shared_ptr. However, to use the proper deleter you must declare your shared_ptr as std::shared_ptr<float[]>, not <float*>.

See more explanations and examples here: https://stackoverflow.com/a/13062069/4477684

Phil, what C++ standard version does Oboe require? If it's C++17, you can go an easy way, otherwise there is more technical work to do.

Reply:
Why is this a `shared_ptr` at all? Isn't `SampleBuffer` the sole owner of the sample data? If so, you can change to `make_unique<float[]>(mNumSamples)`.

Reply:
I'm going to put this off until after this merge. I will seek more guidance from the template gods after that.